src/main/java/wolox/training/models/Book.java

```diff
@@ -84,14 +85,16 @@ public String getGenre() {
     }
 
     public void setGenre(String genre) {
+        Preconditions.checkNotNull(genre);
```

Comment:
The method `checkNotNull(T element)` returns an object of type `T`, I think you should be able to do something like this:

`this.genre = Preconditions.checkNotNull(genre);` which is a bit cleaner :) 

The same goes for all uses of the method (but unfortunately it doesn't apply to the method `checkArgument`)